eth2util/rlp/rlp.go

```diff
@@ -101,9 +101,9 @@ func decodeLength(item []byte) (offset int, length int, err error) {
 	}
 
 	if prefix < 0xc0 {
-		length = int(prefix - 0xb7)
-		if length > 63 || length < 0 {
-			return 0, 0, errors.New("invalid length prefix")
+		length = int(prefix - 0xb7)    // length of the string in bytes in binary form
+		if length > 8 || length <= 0 { // This is impossible based on outer if else checks
```

Comment:
Reasoning:
```
Since
0xb8 <= prefix < 0xc0
=> 184 <= prefix < 192 
=> 184-183 <= prefix-0xb7 < 192-183
=> 1 <= length < 9
=> length range [1,8]
```